pkg/generate/app/cmd/newapp_test.go

```diff
@@ -1066,9 +1182,73 @@ func TestRunBuilds(t *testing.T) {
 				originNamespace: "default",
 			},
 			expected: map[string][]string{
-				"buildConfig": {"origin-base"},
+				"buildConfig": {"foobar"},
 				"imageStream": {"origin-base"},
 			},
+			checkResult: func(res *AppResult) error {
+				for _, item := range res.List.Items {
+					switch t := item.(type) {
+					case *buildapi.BuildConfig:
+						got := t.Spec.Output.To
+						want := &kapi.ObjectReference{
+							Kind: "DockerImage",
+							Name: "destination/reference:tag",
+						}
+						if !reflect.DeepEqual(got, want) {
+							return fmt.Errorf("build.Spec.Output.To = %v; want %v", got, want)
+						}
+						return nil
+					}
+				}
+				return fmt.Errorf("BuildConfig not found; got %v", res.List.Items)
+			},
+		},
+		{
+			name: "successful build from dockerfile with identical input and output image references with warning",
+			config: &AppConfig{
+				Dockerfile: "FROM centos\nRUN yum install -y httpd",
+				To:         "centos",
+
+				dockerSearcher: dockerSearcher,
+				imageStreamSearcher: app.ImageStreamSearcher{
+					Client:            &client.Fake{},
+					ImageStreamImages: &client.Fake{},
+					Namespaces:        []string{"default"},
+				},
+				imageStreamByAnnotationSearcher: &app.ImageStreamByAnnotationSearcher{
+					Client:            &client.Fake{},
+					ImageStreamImages: &client.Fake{},
+					Namespaces:        []string{"default"},
+				},
+				templateSearcher: app.TemplateSearcher{
+					Client: &client.Fake{},
+					TemplateConfigsNamespacer: &client.Fake{},
+					Namespaces:                []string{"openshift", "default"},
+				},
+
+				detector: app.SourceRepositoryEnumerator{
+					Detectors: source.DefaultDetectors,
+					Tester:    dockerfile.NewTester(),
+				},
+				typer:           kapi.Scheme,
+				osclient:        &client.Fake{},
+				originNamespace: "default",
+			},
+			expected: map[string][]string{
+				"buildConfig": {"centos"},
+				"imageStream": {"centos"},
+			},
+			checkOutput: func(stdout, stderr io.Reader) error {
```

Comment:
As I can see we are not using `stdout` here. Do we really need this argument? If not, then, maybe rename this method to something like `checkStdOut()` to reflect what it's doing?